go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -0,0 +1,285 @@
+		//process Rules into egressFirewallRules for egressFirewall struct
+		efr, err := newEgressFirewallRule(egressFirewallRule, i)
+		if err != nil {
+			klog.Errorf("Cannot create EgressFirewall Rule for destination %s to namespace %s - %v",
+				egressFirewallRule.To.CIDRSelector, egressFirewall.Namespace, err)
+			continue
+
+		}
+		ef.egressRules = append(ef.egressRules, efr)
+	}
+
+	existingNodes, err := oc.kube.GetNodes()
+	if err != nil {
+		klog.Errorf("Unable to add egressfirewall %s, cannot list nodes: %s", egressFirewall.Name, err)
+	}
+	for _, node := range existingNodes.Items {
+		joinSwitch := joinSwitch(node.Name)
+		err = ef.addACLToJoinSwitch(joinSwitch, nsInfo.addressSet.GetHashName())
+		if err != nil {
+			klog.Errorf("%s", err)
+		}
+	}
+
+}
+
+func (oc *Controller) updateEgressFirewall(oldEgressFirewall, newEgressFirewall *egressfirewallapi.EgressFirewall) {
+	if !reflect.DeepEqual(oldEgressFirewall.Spec, newEgressFirewall.Spec) {
+		oc.deleteEgressFirewall(oldEgressFirewall)
+		oc.addEgressFirewall(newEgressFirewall)
+	}
+}
+
+func (oc *Controller) deleteEgressFirewall(egressFirewall *egressfirewallapi.EgressFirewall) {
+	klog.Infof("Deleting egress Firewall %s in namespace %s", egressFirewall.Name, egressFirewall.Namespace)
+
+	nsInfo, err := oc.waitForNamespaceLocked(egressFirewall.Namespace)
+	if err != nil {
+		klog.Errorf("Failed to wait for namespace %s event (%v)",
+			egressFirewall.Namespace, err)
+	}
+	defer nsInfo.Unlock()
+	nsInfo.egressFirewall = false
+
+	stdout, stderr, err := util.RunOVNNbctl("--data=bare", "--no-heading",
+		"--columns=_uuid", "find", "ACL",
+		fmt.Sprintf("external-ids:egressFirewall=%s", egressFirewall.Namespace))
+	if err != nil {
+		klog.Errorf(" error executing create ACL command, stderr: %q, %+v", stderr, err)
+		//none of the elements of the egressfirewall have been deleted
+		nsInfo.egressFirewall = true
+		return
+	}
+	uuids := strings.Fields(stdout)
+	existingNodes, err := oc.kube.GetNodes()
+	if err != nil {
+		klog.Errorf("Error deleting egressFirewall for namespace %s, cannot get nodes to delete ACLS %v",
+			egressFirewall.Namespace, err)
+	}
+	for _, uuid := range uuids {
+		for _, node := range existingNodes.Items {
+			_, stderr, err := util.RunOVNNbctl("remove", "logical_switch",
+				joinSwitch(node.Name), "acls", uuid)
+			if err != nil {
+				klog.Errorf("Remove failed to delete the rule for "+
+					"address_set=%s, stderr: %q (%v)", nsInfo.addressSet.GetHashName(), stderr, err)
+			}
+		}
+	}
+}
+
+func (ef *egressFirewall) addACLToJoinSwitch(joinSwitch, hashedAddressSetName string) error {
+	for _, rule := range ef.egressRules {
+		var match string
+		var action string
+		if rule.access == egressfirewallapi.EgressFirewallRuleAllow {
+			action = "allow"
+		} else {
+			action = "drop"
+		}
+
+		if len(rule.ports) == 0 {
+			match = fmt.Sprintf("match=\"ip4.dst == %s && ip4.src == $%s\"", rule.to.cidrSelector, hashedAddressSetName)
```

Comment:
i take ipv6 was never part of the feature ?

Reply:
I don't see why ipv6 couldn't be part of the feature, I added that capability